### PR TITLE
chore: remove deprecated `centralized logs` resources (containers and volumes)

### PR DESCRIPTION
**Pull request by leoporoli**

### CODE REVIEW

File: cli/cli/helpers/engine_manager/engine_existence_guarantor.go [hunk 1]
```diff
@@ -193,11 +181,11 @@ func (guarantor *engineExistenceGuarantor) VisitContainerRunningButServerNotResp
...
 func (guarantor *engineExistenceGuarantor) VisitRunning() error {
 
-	//TODO this condition is a temporary hack, we should removed it when the centralized logs in Kubernetes Kurtosis Backend is implemented
+	//TODO(centralized-logs-infra-deprecation) remove this code in the future when people don't have centralized logs infra running
 	if guarantor.kurtosisClusterType == resolved_config.KurtosisClusterType_Docker {
 		ctx := context.Background()
-		if err := guarantor.ensureCentralizedLogsComponentsAreRunning(ctx, shouldForceLogsComponentsContainersRestartWhenEngineContainerIsRunning); err != nil {
-			return stacktrace.Propagate(err, "An error occurred ensuring that the centralized logs components are running")
+		if err := guarantor.ensureDestroyDeprecatedCentralizedLogsResources(ctx); err != nil {
+			return stacktrace.Propagate(err, "An error occurred removing the old and deprecated centralized logs resources")
```
Comment (h4ck3rk3y) [inline cli/cli/helpers/engine_manager/engine_existence_guarantor.go]: should this throw an error or just print it?

Comment (h4ck3rk3y) [reply cli/cli/helpers/engine_manager/engine_existence_guarantor.go]: I'd rather just print what failed and tell folks how they can cleanup themselves

File: cli/cli/helpers/engine_manager/engine_existence_guarantor.go [hunk 2]
```diff
@@ -193,11 +181,11 @@ func (guarantor *engineExistenceGuarantor) VisitContainerRunningButServerNotResp
 
 func (guarantor *engineExistenceGuarantor) VisitRunning() error {
 
-	//TODO this condition is a temporary hack, we should removed it when the centralized logs in Kubernetes Kurtosis Backend is implemented
+	//TODO(centralized-logs-infra-deprecation) remove this code in the future when people don't have centralized logs infra running
 	if guarantor.kurtosisClusterType == resolved_config.KurtosisClusterType_Docker {
 		ctx := context.Background()
-		if err := guarantor.ensureCentralizedLogsComponentsAreRunning(ctx, shouldForceLogsComponentsContainersRestartWhenEngineContainerIsRunning); err != nil {
-			return stacktrace.Propagate(err, "An error occurred ensuring that the centralized logs components are running")
+		if err := guarantor.ensureDestroyDeprecatedCentralizedLogsResources(ctx); err != nil {
```
Comment (h4ck3rk3y) [inline cli/cli/helpers/engine_manager/engine_existence_guarantor.go]: hmm looks like that function never errors, maybe this check and the return type isn't needed

File: container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/logs_database_functions/destroy_deprecated_logs_database.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	deprecatedLogsDatabaseVolumeName = "kurtosis-logs-db-vol"
+)
+
+// TODO(centralized-logs-database-deprecation) remove this entire function after enough people are on > 0.68.0
+func DestroyDeprecatedCentralizedLogsDatabase(
+	ctx context.Context,
+	dockerManager *docker_manager.DockerManager,
+) error {
+
+	if err := DestroyLogsDatabase(ctx, dockerManager); err != nil {
```
Comment (h4ck3rk3y) [inline container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/logs_database_functions/destroy_deprecated_logs_database.go:20]: i am assuming this removes the container.

File: container-engine-lib/lib/backend_interface/kurtosis_backend.go [hunk 4]
```diff
@@ -385,5 +385,5 @@ type KurtosisBackend interface {
 
 	// Destroy the centralized logs collector
 	// TODO(centralized-logs-collector-deprecation) remove this once we know people are on > 0.66.0
-	DestroyDeprecatedCentralizedLogsCollectorContainerAndVolume(ctx context.Context) error
+	DestroyDeprecatedCentralizedLogsResources(ctx context.Context) error
```
Comment (h4ck3rk3y) [inline container-engine-lib/lib/backend_interface/kurtosis_backend.go:388]: I don't know how any of this plays with K8S any more. Maybe it doesn't matter.

**Review by h4ck3rk3y — APPROVED** [review]

File: cli/cli/helpers/engine_manager/engine_existence_guarantor.go (hunk 2 above)
Comment (leoporoli) [reply cli/cli/helpers/engine_manager/engine_existence_guarantor.go]: yes, that's right, good catch!

File: container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/logs_database_functions/destroy_deprecated_logs_database.go (hunk 3 above)
Comment (leoporoli) [reply container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/logs_database_functions/destroy_deprecated_logs_database.go:20]: yes, I added a comment to clarify

File: container-engine-lib/lib/backend_interface/kurtosis_backend.go (hunk 4 above)
Comment (leoporoli) [reply container-engine-lib/lib/backend_interface/kurtosis_backend.go:388]: it doesn't work in K8s but I will rename the method in the other repository just to implement the new signature

**Pull request merged**